shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -94,7 +96,13 @@ + (Class)layerClass {
 }
 
 - (std::unique_ptr<flutter::IOSSurface>)createSurface:
-    (std::shared_ptr<flutter::IOSGLContext>)context {
+    (fml::WeakPtr<flutter::IOSGLContext>)resourceContext {
+#if !TARGET_IPHONE_SIMULATOR
+  if (!_onscreenContext) {
```

Comment:
When would this condition be false?
Or - why isn't this `FML_DCHECK(!_onscreenContext)`?

Reply:
it would be false if this is the first time createSurface is called after the FlutterView has been constructed. I put the check in as (I assume) createSurface can be called multiple times during the lifetime of the FlutterView and we don't want to generate new onscreen contexts for each surface.

Reply:
This matches existing behavior  where the platform view would never reset the context. If we change this, we should do it in a separate patch.